mozetl/hardware_report/summarize_json.py

```diff
@@ -0,0 +1,716 @@
+    # then we're on a 64 bit system.
+    if is_windows and is_wow64:
+        return 'x86-64'
+
+    # Otherwise we're probably on a 32 bit system.
+    return 'x86'
+
+
+def vendor_name_from_id(id):
+    """ Get the string name matching the provided vendor id.
+
+    Args:
+        id: A string containing the vendor id.
+
+    Returns:
+        A string containing the vendor name or "(Other <ID>)" if
+        unknown.
+    """
+
+    # TODO: We need to make this an external resource for easier
+    # future updates.
+    vendor_map = {
+        '0x1013': 'Cirrus Logic',
+        '0x1002': 'AMD',
+        '0x8086': 'Intel',
+        '0x5333': 'S3 Graphics',
+        '0x1039': 'SIS',
+        '0x1106': 'VIA',
+        '0x10de': 'NVIDIA',
+        '0x102b': 'Matrox',
+        '0x15ad': 'VMWare',
+        '0x80ee': 'Oracle VirtualBox',
+        '0x1414': 'Microsoft Basic',
+    }
+
+    return vendor_map.get(id, "Other")
+
+
+def get_device_family_chipset(vendor_id, device_id):
+    """ Get the family and chipset strings given the vendor and device ids.
+
+    Args:
+        vendor_id: a string representing the vendor id (e.g. '0xabcd').
+        device_id: a string representing the device id (e.g. '0xbcde').
+
+    Returns:
+        A string in the format "Device Family Name-Chipset Name".
+    """
+    if not vendor_id in device_map:
+        return "Unknown"
+
+    if not device_id in device_map[vendor_id]:
+        return "Unknown"
+
+    return "-".join(device_map[vendor_id][device_id])
+
+
+def invert_device_map(m):
+    """ Inverts a GPU device map fetched from the jrmuizel's Github repo.
+    The layout of the fetched GPU map layout is:
+        Vendor ID -> Device Family -> Chipset -> [Device IDs]
+    We should convert it to:
+        Vendor ID -> Device ID -> [Device Family, Chipset]
+    """
+    device_id_map = {}
+    for vendor, u in m.iteritems():
+        device_id_map['0x' + vendor] = {}
+        for family, v in u.iteritems():
+            for chipset, ids in v.iteritems():
+                device_id_map['0x' +
+                              vendor].update({('0x' +
+                                               gfx_id): [family, chipset] for gfx_id in ids})
+    return device_id_map
+
+
+def build_device_map():
+    """ This function builds a dictionary that will help us mapping vendor/device ids to a
+    human readable device family and chipset name."""
+
+    intel_raw = fetch_json(
+        "https://github.com/jrmuizel/gpu-db/raw/master/intel.json")
+    nvidia_raw = fetch_json(
+        "https://github.com/jrmuizel/gpu-db/raw/master/nvidia.json")
+    amd_raw = fetch_json(
+        "https://github.com/jrmuizel/gpu-db/raw/master/amd.json")
+
+    device_map = {}
+    device_map.update(invert_device_map(intel_raw))
+    device_map.update(invert_device_map(nvidia_raw))
+    device_map.update(invert_device_map(amd_raw))
+
+    return device_map
+
+
+device_map = build_device_map()
```

Comment:
let's move this down to main() (or whever it gets used, even better) and supply it as a argument rather than using it globally.